stormlog/_wandb/attribution.py

```
@@ -0,0 +1,226 @@
+            summary_fields["stormlog_attribution_history_recorded"] = history_recorded
+
+    return summary_fields
+
+
+def build_attribution_preview_html(root: Path) -> str:
+    snapshot_path = root / SNAPSHOT_PICKLE_FILENAME
+    tensor_index = read_json_if_exists(root / TENSOR_ATTRIBUTION_FILENAME)
+    if snapshot_path.exists() and isinstance(tensor_index, Mapping):
+        try:
+            with snapshot_path.open("rb") as handle:
+                snapshot = pickle.load(handle)
```

Ruff (0.15.10)

[error] `pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)